src/components/SharedComponents/LocationDisplay.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import { Body4 } from "components/SharedComponents";
+import { View } from "components/styledComponents";
+import * as React from "react";
+import { useTranslation } from "react-i18next";
+import IconMaterial from "react-native-vector-icons/MaterialIcons";
+import useLocationName from "sharedHooks/useLocationName";
+
+type Props = {
+  latitude: number,
+  longitude: number
+};
+
+const LocationDisplay = ( {
+  latitude,
+  longitude
+}: Props ): React.Node => {
+  const { t } = useTranslation( );
+  const locationName = useLocationName( latitude, longitude );
+
+  return (
+    <View className="flex flex-row items-center">
+      <IconMaterial name="location-pin" size={15} />
+      <Body4 className="text-darkGray ml-[5px]">
+        {locationName || t( "no place guess" )}
+      </Body4>
+    </View>
+  );
+};
+
+export default LocationDisplay;
```

Comment:
Since this does need to be specific to showing the location of an observation, let's rename it to `ObservationLocation` or `ObsLocationDisplay` or something that makes it clear this is just for observations.